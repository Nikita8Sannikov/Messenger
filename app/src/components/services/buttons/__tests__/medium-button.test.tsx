import { render, screen } from "@testing-library/react";

import MediumButtonComponent from "@components/services/buttons/medium-button";

describe("MediumButtonComponent", () => {
	test("renders with children", () => {
		render(<MediumButtonComponent>Test Button</MediumButtonComponent>);
		const button = screen.getByText("Test Button");
		expect(button).toBeInTheDocument();
	});

	test("applies medium size prop", () => {
		render(<MediumButtonComponent>Test Button</MediumButtonComponent>);
		const button = screen.getByText("Test Button");
		expect(button).toHaveClass("MuiButton-sizeMedium");
	});

	test("passes through other props correctly", () => {
		render(
			<MediumButtonComponent 
				variant="outlined" 
				color="secondary"
				disabled
				className="custom-class"
			>
                Test Button
			</MediumButtonComponent>,
		);
		const button = screen.getByText("Test Button");
		expect(button).toHaveClass("MuiButton-outlined");
		expect(button).toHaveClass("MuiButton-colorSecondary");
		expect(button).toHaveClass("custom-class");
		expect(button).toBeDisabled();
	});

	test("handles click events", () => {
		const handleClick = jest.fn();
		render(
			<MediumButtonComponent onClick={handleClick}>
                Test Button
			</MediumButtonComponent>,
		);
		const button = screen.getByText("Test Button");
		button.click();
		expect(handleClick).toHaveBeenCalledTimes(1);
	});

	test("matches snapshot", () => {
		const { asFragment } = render(
			<MediumButtonComponent variant="outlined" color="secondary">
                Test Button
			</MediumButtonComponent>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
