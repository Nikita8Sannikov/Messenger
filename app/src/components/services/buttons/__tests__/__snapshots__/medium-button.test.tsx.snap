// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediumButtonComponent matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary css-1ioaic5-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Test Button
  </button>
</DocumentFragment>
`;
